youtube_dl/extractor/laola1tv.py

```
@@ -1,86 +1,114 @@
+        _v = lambda x, **k: xpath_text(hd_doc, './/video/' + x, **k)
+        title = _v('title', fatal=True)
 
-        categories = xpath_text(hd_doc, './/video/meta_sports')
+        categories = _v('meta_sports')
         if categories:
             categories = categories.split(',')
 
-        ident = random.randint(10000000, 99999999)
-        token_url = '%s&ident=%s&klub=0&unikey=0&timestamp=%s&auth=%s' % (
-            flash_url, ident, flashvars['timestamp'], flashvars['auth'])
+        time_date = _v('time_date')
+        time_start = _v('time_start')
+        upload_date = None
+        if time_date and time_start:
+            upload_date = unified_strdate(time_date + ' ' + time_start)
+
+        json_url = ('https://club.laola1.tv/sp/laola1/api/v3/user/session' +
+                    '/premium/player/stream-access?videoId=%s&target=2' +
+                    '&label=laola1tv&area=%s') % (video_id, _v('area'))
+        req = sanitized_Request(json_url, urlencode_postdata(
+            dict((i, v) for i, v in enumerate(_v('req_liga_abos').split(',')))))
 
+        token_url = self._download_json(req, video_id)['data']['stream-access'][0]
         token_doc = self._download_xml(
             token_url, video_id, note='Downloading token')
-        token_attrib = token_doc.find('.//token').attrib
-        if token_attrib.get('auth') in ('blocked', 'restricted'):
-            raise ExtractorError(
-                'Token error: %s' % token_attrib.get('comment'), expected=True)
 
-        video_url = '%s?hdnea=%s&hdcore=3.2.0' % (
-            token_attrib['url'], token_attrib['auth'])
+        token_attrib = xpath_element(token_doc, './/token').attrib
+        token_auth = token_attrib['auth']
+        video_url = token_attrib['url']
+
+        if token_auth in ('blocked', 'restricted'):
+            raise ExtractorError(
+                'Token error: %s' % token_attrib['comment'], expected=True)
+        elif token_auth and video_url:
```

If will fail anyway if any of these fields is missing and it's better to fail early explicitly in extractor than in some generic code.